src/party.cpp

```diff
@@ -12,11 +12,11 @@
 extern Game g_game;
 extern Events* g_events;
 
-Party::Party(Player* leader) : leader(leader) { leader->setParty(this); }
+Party::Party(Player* leader) : leader(leader) { leader->setParty(shared_from_this()); }
```

Comment:
shared_from_this() is not available in constructor, so this will throw exception

Reply:
Also, arguably `setParty` is responsibility of the function that creates the party.

Reply:
There should be clear ownership which is now unknown, we could say its the leader who owns a party, but this may lead to crashes if leader is deleted ex on logout. Imo this should follow the pattern and store every created party in game class, party class should have static factory method such as createParty (similar to createNpc)